lib/cdo/delete_accounts_helper.rb

```diff
@@ -0,0 +1,186 @@
+      CDO.sources_s3_bucket => CDO.sources_s3_directory
+    }.each do |bucket, directory|
+      unless bucket.present? && directory.present?
+        raise "Missing AWS s3 bucket information (bucket: #{bucket}, directory: #{directory})."
+      end
+      s3_client.buckets[bucket].objects.with_prefix("/#{directory}/#{storage_id}/").delete_all
+    end
+
+    # Query the DB for the channel IDs associated with the storage ID, deleting data on Firebase
+    # for each.
+    PEGASUS_DB[:storage_apps].where(storage_id: storage_id).each do |storage_app|
+      encrypted_channel_id = storage_encrypt_channel_id storage_id, storage_app[:id]
+      # TODO(asher): This makes more sense as
+      #   FirebaseHelper.new.delete_channel(encrypted_channel_id).
+      # Refactor FirebaseHelper to allow this.
+      FirebaseHelper.new(encrypted_channel_id).delete_channel
+    end
+  end
+
+  # Cleans all LevelSource-backed progress associated with the user.
+  # WARNING: Since the relationship between user_levels and activities and level_sources is
+  #   many-to-one, we cannot be certain that this is not messing with the integrity of the data for
+  #   other users.
+  # @param [Integer] user_id The user to clean the LevelSource-backed progress of.
+  def self.clean_level_source_backed_progress(user_id)
+    GalleryActivity.where(user_id: user_id).each do |gallery_activity|
+      gallery_activity.level_source.try(:clear_data_and_image)
+    end
+    GalleryActivity.where(user_id: user_id).destroy_all
+
+    UserLevel.where(user_id: user_id).find_each do |user_level|
+      if OPEN_ENDED_LEVEL_TYPES.include? user_level.level.type
+        user_level.level_source.try(:clear_data_and_image)
+      end
+    end
+
+    Activity.where(user_id: user_id).find_each do |activity|
+      if OPEN_ENDED_LEVEL_TYPES.include? activity.level.type
+        activity.level_source.try(:clear_data_and_image)
+      end
+    end
+
+    # Note that the `overflow_activities` table exists only in the production environment.
+    if ActiveRecord::Base.connection.data_source_exists? 'overflow_activities'
+      OverflowActivity.where(user_id: user_id).find_each do |activity|
+        if OPEN_ENDED_LEVEL_TYPES.include? activity.level.type
+          activity.level_source.try(:clear_data_and_image)
+        end
+      end
+    end
+  end
+
+  # Cleans the responses for all surveys associated with the user.
+  # @param [Integer] user_id The user to clean the surveys of.
+  def self.clean_survey_responses(user_id)
+    SurveyResult.where(user_id: user_id).each(&:clear_open_ended_responses)
+  end
+
+  # Purges all student users whose acceptance of our Terms of Service and Privacy Policy
+  # is only through the being purged user.
+  # NOTE: This method assumes the sections and followers associated with user_id have already been
+  # destroyed.
+  # @param [Integer] user_id for which to delete orphaned students.
+  def self.purge_orphaned_students(user_id)
```

Comment:
Feel welcome to take the issue up in the [spec](https://docs.google.com/document/d/1ChqYFYusg4lnKK7_q2Wq9x1ipRrYs4E5lL9pccRrWNY) or with @poorvasingal directly.